reaper/reaper.go

```diff
@@ -17,141 +17,153 @@
 package reaper
 
 import (
+	"errors"
 	"fmt"
-	"io/ioutil"
+	"github.com/pivotal-cf/service-instance-reaper/cloudfoundry"
+	"io"
 	"time"
-	"github.com/pivotal-cf/service-instance-reaper/httpclient"
-	"net/http"
-	"encoding/json"
 )
 
-// FIXME: support multi-page responses
-const MAXIMUM_RESULTS_PER_PAGE = 100
+const MaximumResultsPerPage = 50
 
-func Reap(authClient httpclient.AuthenticatedClient, apiUrl string, serviceName string, accessToken string, expiryInterval time.Duration, reap bool, fatal func(string)) {
-	bodyReader, statusCode, err := authClient.DoAuthenticatedGet(apiUrl+fmt.Sprintf("/v2/services?q=label:%s", serviceName), accessToken)
-	if err != nil {
-		fatalError("GET /v2/services failed", err, fatal)
-	}
-	if statusCode != http.StatusOK {
-		fatal(fmt.Sprintf("GET /v2/services failed: %d", statusCode))
-	}
+type reaper struct {
+	cf             cloudfoundry.Client
+	expiryInterval time.Duration
+	reap           bool
+	recursive      bool
+	currentTime    func() time.Time
+	output         io.Writer
+	errorChannel   chan error
+}
 
-	if bodyReader == nil {
-		fatal("GET /v2/services response body missing")
+func Reap(cf cloudfoundry.Client, serviceName string, expiryInterval time.Duration, reap, recursive bool, currentTime func() time.Time, output io.Writer) error {
```

Comment:
The function signature of `Reap()` could be improved if we used a receiver and put the things we want to mock there